src/teacher.py

```diff
@@ -75,6 +80,81 @@ def _answer_Demonstration_query(
+        # goals (and are not necessarily associated with any particular task).
+        trajectory = None
+        null_response = DemonstrationResponse(query, teacher_traj=None)
+        goal_state = query.goal_state
+        if CFG.env == "cover_multistep_options":
+            assert CFG.cover_multistep_use_learned_equivalents
+            # Setup.
+            block_type, target_type, robot_type = _get_types_by_names(
+                CFG.env, ["block", "target", "robot"])
+            Pick, Place = _get_options_by_names(
+                CFG.env, ["LearnedEquivalentPick", "LearnedEquivalentPlace"])
+            robot = state.get_objects(robot_type)[0]
+            blocks = state.get_objects(block_type)
+            targets = state.get_objects(target_type)
+            state_blocks_held = [
+                b for b in blocks if abs(state.get(b, "grasp") - 1) < 1e-6
+            ]
+            goal_blocks_held = [
+                b for b in blocks if abs(goal_state.get(b, "grasp") - 1) < 1e-6
+            ]
+            # Case 1: Pick.
+            if len(goal_blocks_held) == 1:
+                parameterized_option = Pick
+                block = goal_blocks_held[0]
+                arguments = [block, robot]
+                changing_objs = [block, robot]
+            # Case 2: Place.
+            elif len(state_blocks_held) == 1:
+                parameterized_option = Place
+                block = state_blocks_held[0]
+                # The target does not actually matter, because it is not
+                # involved in the definition of the Place policy. The only
+                # reason that it's included in the ParameterizedOption is
+                # that there is a change in the symbolic effects for the
+                # target (it becomes covered).
+                target = targets[0]
+                arguments = [block, robot, target]
+                changing_objs = [block, robot]
+            # Case 3: invalid or unsupported request.
+            else:
+                return null_response
+            params = goal_state.vec(changing_objs) - state.vec(changing_objs)
+            option = parameterized_option.ground(arguments, params)
+            policy = utils.option_plan_to_policy([option])
+            termination_function = lambda s: s.allclose(goal_state)
+            trajectory = utils.run_policy_with_simulator(
+                policy,
+                self._simulator,
+                state,
+                termination_function,
+                max_num_steps=CFG.max_num_steps_option_rollout)
+        else:
+            raise NotImplementedError("State-based demonstration queries not"
+                                      f"supported for env {CFG.env}.")
+        # Validate. If the goal state was not reached, the query is assumed
+        # to be invalid, and a None trajectory is returned.
+        if trajectory is None or not trajectory.states[-1].allclose(
```

Comment:
what code path could possibly cause `trajectory` to be None here?

Reply:
good point, none, will remove